File: 3498.reverse-degree-of-a-string.java
/*
 * @lc app=leetcode id=3498 lang=java
 *
 * [3498] Reverse Degree of a String
 */

// @lc code=start
class Solution {
    public int reverseDegree(String s) {
    int res = 0;
    char[] ch = s.toCharArray();
    for (int i = 0; i < ch.length; i++) {
        res += (i+1) * (26 - (ch[i] - 'a'));
    }
    return res;
        
    }
}

// @lc code=end

File: 3498.reverse-degree-of-a-string.py
#
# @lc app=leetcode id=3498 lang=python
#
# [3498] Reverse Degree of a String
#

# @lc code=start
class Solution(object):
    def reverseDegree(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        for i in range(len(s)):
            res += (i + 1) * (26 - (ord(s[i]) - ord('a')))
        return res
        
# @lc code=end

